command/server/config.go

```diff
@@ -151,6 +151,67 @@ ui = true
 	return parsed, nil
 }
 
+// DevTLSConfig is a Config that is used for dev tls mode of Vault.
+func DevTLSConfig(storageType string) (*Config, string, error) {
+	dir, err := os.MkdirTemp("", "vault-tls")
+	if err != nil {
+		return nil, "", err
+	}
+
+	ca, err := GenerateCA()
+	if err != nil {
+		return nil, "", err
+	}
+
+	cert, key, err := GenerateCert(ca.Template, ca.Signer)
+	if err != nil {
+		return nil, "", err
+	}
+
+	if err := os.WriteFile(fmt.Sprintf("%s/vault-ca.pem", dir), []byte(ca.PEM), 0o644); err != nil {
+		return nil, "", err
+	}
+
+	if err := os.WriteFile(fmt.Sprintf("%s/vault-cert.pem", dir), []byte(cert), 0o644); err != nil {
```

Comment:
Ideally we'd restrict access to the vault cert and key more than the CA. I do agree that the CA needs to be wider permissions in general (if the dev server is running as a different user), but it'd really be nice to restrict the cert/key down.

Reply:
I lowered them to 0444, 0400, 0400 respectively.